### PR TITLE
Live view script added for galvo phase and gain alignment

### PR DESCRIPTION
**Pull request by tepals**
- fit a sine to the image and display this on the image

### CODE REVIEW

File: scripts/galvo_phase_gain_calibration.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+from odemis.util.driver import get_backend_status, BACKEND_RUNNING
+
+PIXEL_SIZE_UM = 3.45 / 50.
+
+
+class VideoDisplayerSine(VideoDisplayer):
+    """
+    Very simple display for a continuous flow of images as a window with an overlay of a sine wave.
+    It should be pretty much platform independent.
+    """
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/galvo_phase_gain_calibration.py]: VideoDisplayer is the base class of VideoDisplayerSine, so it's not a parameter and therefore not added as a parameter to the docstring. You usually don't add base classes to the docstring

File: scripts/galvo_phase_gain_calibration.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        """
+        Displays the window on the screen
+        size: 2-tuple int,int
+            X and Y size of the window at initialisation.
+            Note that the size of the window automatically adapts afterwards to the
+            coming pictures
+        """
+        self.app = ImageWindowApp(title, size)
+
+    def new_image(self, data):
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/galvo_phase_gain_calibration.py]: This overwrites `new_image` if the `VideoDisplayer` class, therefore should not be renamed here. Added this to the docstring.

File: scripts/galvo_phase_gain_calibration.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        im_passer.display = False
+        im_passer.available.set()  # Force the thread to check the .display flag
+        dataflow.unsubscribe(new_image_wrapper)
+        if kill_ccd:
+            ccd.terminate()
+
+
+def fit_sine_to_image(image):
+    """
+    Fit a sine wave to an image containing a sine somewhere in the image.
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/galvo_phase_gain_calibration.py]: no

[the remark this answers is not in the document]
Comment (Sabrina85) [reply scripts/galvo_phase_gain_calibration.py]: In Odemis we often add the shape the image needs to have. I would suggest to add below in the params the shape of the input image.

File: scripts/galvo_phase_gain_calibration.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        for text in info:
+            ctx2.translate(0, font_size)
+            ctx2.show_text(text)
+            ctx2.stroke()
+
+        if wx.MAJOR_VERSION <= 3:
+            self.imageCtrl.SetBitmap(wx.BitmapFromImage(self.img))
+            self.imageCtrlSine.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(point_surface))
+            self.imageCtrlText.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(text_surface))
+        else:
```
Comment (pieleric) [inline scripts/galvo_phase_gain_calibration.py]: In general, you can drop the support for wxPython 3. With Odemis 2.10, wxPython 4 is the default version, and in Ubuntu 18.04, wxPython 3 is not even available.

File: scripts/galvo_phase_gain_calibration.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+        """
+        Update the window with the new image and fit a sine to the image.
+        This overwrites new_image of the VideoDisplayer class.
+        data: numpy.ndarray
+            A 2D array containing the image (can be 3D if in RGB)
+        """
+        self.app.params, self.app.cols = fit_sine_to_image(data)
+        super(VideoDisplayerSine, self).new_image(data)
+
+    def waitQuit(self):
```
Comment (pieleric) [inline scripts/galvo_phase_gain_calibration.py]: Drop this method, as it's the same as the parent's one.

File: scripts/galvo_phase_gain_calibration.py [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+        Dataflow object to access.
+    kill_ccd: bool
+        True if it is required to terminate the ccd after closing the window.
+    """
+    # create a window
+    window = VideoDisplayerSine("Live from %s.%s" % (ccd.role, "data"), ccd.resolution.value)
+    im_passer = ImagePasser()
+    t = threading.Thread(target=image_update, args=(im_passer, window))
+    t.daemon = True
+    t.start()
```
Comment (pieleric) [inline scripts/galvo_phase_gain_calibration.py]: It should be possible to integrate all of this into VideoDisplayerSine: it creates the thread, and all the attributes of "ImagePasser" are directly part of VideoDisplayerSine.
Then the new_image_wrapper just has to pass the image to a new method `VideoDisplayerSine.new_image_and_just_store_it_for_next_time_you_have_time_to_process_it()`. (or a better name ;-) )

File: scripts/galvo_phase_gain_calibration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pieleric) [reply scripts/galvo_phase_gain_calibration.py]: I agree with Sabrina, if you just extend the image description with `shape of (Y,X)`, it'll clarify a lot the input type.

**Comment by tepals** [comment]
@pieleric updated and cleaned up the image thread

**Pull request merged**